src/components/modals/SurveyModal.tsx

```diff
@@ -24,71 +26,59 @@ import { EdgeModal } from './EdgeModal'
-  { catKey: 'referral', label: lstrings.survey_opt_personal_referral, selected: false },
-  { catKey: 'search', label: lstrings.survey_opt_search_engine, selected: false },
-  { catKey: 'social', label: lstrings.survey_opt_youtube, selected: false }
+  { catKey: 'ads', label: lstrings.survey_opt_ad },
+  { catKey: 'appStore', label: lstrings.survey_opt_app_store },
+  { catKey: 'article', label: lstrings.survey_opt_article },
+  { catKey: 'inPerson', label: lstrings.survey_opt_in_person_event },
+  { catKey: 'referral', label: lstrings.survey_opt_personal_referral },
+  { catKey: 'search', label: lstrings.survey_opt_search_engine },
+  { catKey: 'social', label: lstrings.survey_opt_youtube }
 ]
 
 interface LocaleSubcategory {
   [langKey: string]: string
 }
 
-// TODO: Update info server with finalized type
-const SURVEY_SUBCATS_TEMP: { [catKey: string]: LocaleSubcategory[] } = {
-  ads: [{ en: 'Social media ad (YouTube, X, Instagram, etc.)' }, { en: 'Website display/banner ad' }, { en: 'Search engine ad' }, { en: 'Podcast ad' }],
-  appStore: [{ en: 'Featured in the app store' }, { en: 'Search result in the app store' }],
-  article: [{ en: 'Crypto-specific blog' }, { en: 'General tech blog' }, { en: 'News outlet' }, { en: 'Influencer blog' }],
-  inPerson: [{ en: 'Conference/trade show' }, { en: 'Meetup' }, { en: 'Hackathon' }, { en: 'Crypto Canal', fr: 'Le Canal Crypto' }],
-  referral: [{ en: 'Friend or family recommendation' }, { en: 'Online forum discussion' }, { en: 'Community group (e.g., Discord, Telegram)' }],
-  search: [{ en: 'Google' }, { en: 'Bing' }, { en: 'DuckDuckGo' }, { en: 'Brave' }],
-  social: [
-    { en: 'X (Twitter)' },
-    { en: 'YouTube' },
-    { en: 'Instagram' },
-    { en: 'LinkedIn' },
-    { en: 'Reddit' },
-    { en: 'TikTok' },
-    { en: 'Facebook' },
-    { en: 'Cris Cyborg' }
-  ]
-}
+type SurveyOption2Keys = keyof InstallSurvey2['surveyOptions2']
 
 export const SurveyModal = (props: { bridge: AirshipBridge<void> }) => {
   const { bridge } = props
   const dispatch = useDispatch()
   const theme = useTheme()
   const styles = getStyles(theme)
 
-  // Randomize response options, excluding "Other"
-  const [options, setOptions] = useState([...shuffleArray(SURVEY_CATS), { catKey: 'other', label: lstrings.survey_opt_other_specify, selected: false }])
-  const [selectedIndex, setSelectedIndex] = useState<number>()
-
-  // "Other" custom text response
-  const [otherText, setOtherText] = useState('')
+  /**
+   * Shuffle the categories, then append an “other” category at the end:
+   */
+  const [options] = React.useState<SurveyCategory[]>(() => [...shuffleArray(SURVEY_CATS), { catKey: 'other', label: lstrings.survey_opt_other_specify }])
+
+  /**
+   * Store the user's selection here (rather than on each item).
+   * This can be `undefined`, or one of the category keys (like "ads", "other", etc.)
+   */
+  const [selectedCatKey, setSelectedCatKey] = React.useState<string | undefined>()
```

Comment:
It could be `SurveyOption2Keys | 'other' | undefined` instead of `string | undefined`. That would eliminate the `as SurveyOption2Keys` below in `handleNextPress`.

Making this change might require other place to tighten up their types too. This could be annoying, so if you can't get it done in 15 minutes of effort or so, just leave it be.

Reply:
Yeah I tinkered with this earlier and had some trouble, should have just added
"other" into the info server type in hindsight